libs/SalesforceSDKCore/SalesforceSDKCore/Classes/RestAPI/SFRestRequest.m

```diff
@@ -122,84 +126,83 @@ - (void)setCustomRequestBodyStream:(NSInputStream* (^)(void))bodyInputStreamBloc
 # pragma mark - send and cancel
 
 - (NSURLRequest *)prepareRequestForSend:(SFUserAccount *)user {
-    if (user) {
-
-        /*
-         * If an absolute URL is passed in, use it as-is. If a relative URL is passed in,
-         * parse it and put the pieces together to construct the full URL.
-         */
-        NSMutableString *fullUrl = nil;
-        if ([[self.path lowercaseString] hasPrefix:@"https://"]) {
```

Comment:
@trooper2013 We still need this `if` condition as I mentioned in our conversation offline. There are a few different use cases for this. Some are covered by your new method that takes a custom endpoint (the native use cases), but removing this will break existing clients that rely on this behavior (we have some users internally). Also, it will remove the ability to accomplish this from our `Cordova` and `ReactNative` apps. We need to bring this `if` condition back, at least for now.